Sprint_2

Gomina opened the pull request:
Простите, в первом пулреквесте, что-то пошло не так.
Перезалила файлы.

Gomina committed:
Commit_1
Сделаны тесты для BooksCollector
Добавлено описание тестов в файл README

Gomina committed:
Commit_2
исправлены ошибки в файле tests.py

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,15 @@
+        'Ходячий замок': 'Фантастика',
+        'Легенды ночных стражей': None
+    }
+    yield collector
```
Можно лучше: если в фикстуре после yield ничего не выполняется, то правильнее будет использовать return

ShamilovR commented on tests.py:
```diff
@@ -1,24 +1,142 @@
+                            ])
+
+    def test_add_new_book_with_different_title(self,name, expected_result):
+        collector = BooksCollector()
```
Можно лучше: создание объекта BooksCollector можно вынести в фикстуры

ShamilovR commented on tests.py:
```diff
@@ -1,24 +1,142 @@
+    def test_add_new_book_with_different_title(self,name, expected_result):
+        collector = BooksCollector()
+        collector.add_new_book(name)
+        actual_result = len(collector.books_genre) > 0
+        assert actual_result == expected_result
```
Можно лучше: правильнее проверять конкретное значение в коллекции, а не количество элементов в этой коллекции

ShamilovR commented on tests.py:
```diff
@@ -1,24 +1,142 @@
+        collector.set_book_genre(name, genre)
+        actual_result = name in collector.books_genre and collector.books_genre[name] == genre
+        assert actual_result == expected_result
+
```
Нужно исправить: отсутствует проверка метода get_book_genre

Gomina committed:
Commit_3
В файле conftest.py  yield изменен на return

Gomina committed:
Commit_4
В файл tests.py добавлен тест test_get_book_genre на проверку метода get_book_genre
В файл README.md добавлено описание теста test_get_book_genre

Gomina committed:
Commit_5
Создана фикстура collector и добавлена в тесты
